src/main/scala/edu/ie3/simona/ontology/messages/flex/MinMaxFlexibilityMessage.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * © 2024. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.simona.ontology.messages.flex
+
+import edu.ie3.simona.exceptions.CriticalFailureException
+import edu.ie3.simona.ontology.messages.flex.FlexibilityMessage.ProvideFlexOptions
+import squants.Power
+
+import java.util.UUID
+
+/** Messages that communicate interval-based flexibility with minimum, reference
+  * and maximum power
+  */
+object MinMaxFlexibilityMessage {
+
+  /** Message that provides flexibility options using reference, minimum and
+    * maximum power. It is possible that the power values are either all
+    * negative or all positive, meaning that feed-in or load is mandatory.
+    *
+    * @param modelUuid
+    *   The UUID of the flex provider asset model
+    * @param ref
+    *   The reference active power that the flex provider would produce/consume
+    *   regularly at the current tick, i.e. if it was not flex-controlled
+    * @param min
+    *   The minimum active power that the flex provider allows at the current
```

Comment:
I introduced this term here because besides system participants, also EmAgents can provide flex options to their respective parent EmAgent.
Maybe `flex options provider` is more specific here?